mungegithub/mungers/submit-queue.go

```diff
@@ -543,18 +553,34 @@ func (sq *SubmitQueue) EachLoop() error {
 }
 
 // RegisterOptions registers options used by the SubmitQueue.
-func (sq *SubmitQueue) RegisterOptions(opts *options.Options) {
+func (sq *SubmitQueue) RegisterOptions(opts *options.Options) sets.String {
+	sq.opts = opts
 	opts.RegisterStringSlice(&sq.NonBlockingJobNames, "nonblocking-jenkins-jobs", []string{}, "Comma separated list of jobs that don't block merges, but will have status reported and issues filed.")
 	opts.RegisterBool(&sq.FakeE2E, "fake-e2e", false, "Whether to use a fake for testing E2E stability.")
 	opts.RegisterStringSlice(&sq.DoNotMergeMilestones, "do-not-merge-milestones", []string{}, "List of milestones which, when applied, will cause the PR to not be merged")
 	opts.RegisterInt(&sq.AdminPort, "admin-port", 9999, "If non-zero, will serve administrative actions on this port.")
 	// If you create a StringSliceVar you may wish to check out 'cleanStringSlice()'
-	opts.RegisterString(&sq.Metadata.HistoryUrl, "history-url", "", "URL to access the submit-queue instance's health history.")
-	opts.RegisterString(&sq.Metadata.ChartUrl, "chart-url", "", "URL to access the submit-queue instance's health charts.")
+	opts.RegisterString(&sq.Metadata.historyUrl, "history-url", "", "URL to access the submit-queue instance's health history.")
+	opts.RegisterString(&sq.Metadata.chartUrl, "chart-url", "", "URL to access the submit-queue instance's health charts.")
 	opts.RegisterString(&sq.BatchURL, "batch-url", "", "Prow data.json URL to read batch results")
 	opts.RegisterString(&sq.ContextURL, "context-url", "", "URL where the submit queue is serving - used in Github status contexts")
 	opts.RegisterBool(&sq.GateApproved, "gate-approved", false, "Gate on approved label")
 	opts.RegisterBool(&sq.GateCLA, "gate-cla", false, "Gate on cla labels")
+
+	return sets.NewString(
+		"batch-url",   // Need to start or kill batch processing.
+		"context-url", // Need to remunge all PRs to update statuses with new url.
```

Comment:
Yes, if we change a gate from false to true I don't think there are any problems, but if the gate changes from true to false then we need to remunge all PRs. fixed.